clang/lib/AST/ExprCXX.cpp

```diff
@@ -166,6 +166,55 @@ QualType CXXTypeidExpr::getTypeOperand(ASTContext &Context) const {
       Operand.get<TypeSourceInfo *>()->getType().getNonReferenceType(), Quals);
 }
 
+namespace {
+static bool isGLValueFromPointerDeref(const Expr *E) {
```

Comment:
Pretty sure `namespace {` *and* `static` is redundant; we usually only use `static`.